### ISSUE TITLE
Manage folders alternates between two sets of folders upon refresh

### ISSUE THREAD

**Issue by unode** [k9mail/k-9#4757]
### Expected behavior
Refreshing folder list should display all folders on the server

### Actual behavior
The client alternates between two sets of folders.
Folders that exist when the account was added to k9-mail and folders that were created/renamed afterwards.
With the first refresh I see all the "old" folders on the server. After the refresh I see only new folders and the old folders are no longer accessible through the folder navigation side-menu.

The refresh alternates between showing 7 folders, including `Inbox` and `Outbox` or 50 folders including `Inbox` and `Outbox`. `Inbox` and `Outbox` are the only folders that are always visible between refreshes.

### Steps to reproduce
I'm not sure these will reproduce (i.e. I didn't test) but based on my diagnostic above
1. Add a new account to k9-mail
2. Create new folders on the mail server by using a different client (Thunderbird in my case)
3. Go to "manage folders" and refresh the list

### Environment
K-9 Mail version: 5.713
Android version: 9
Account type: IMAP

### Logs
I have omitted some folder names from the log for privacy reasons.
[k9-log.txt](https://github.com/k9mail/k-9/files/4611112/k9-log.txt)

The log shows:
```
K9StorageEditor: Preferences commit took 19 ms
K9StorageEditor: Committing preference changes
```
These messages repeat in alternating numbers that seem to match the number of folders displayed.

**Comment by cketti** [comment]
I'm not sure I follow. Can you please add a step by step description of what you see (maybe provide screenshots) and what actions you perform?

Are you sure you're not changing which folders to display under 'Manage folders > Show folders…' or 'Account settings > Folders > Folders to display'?

**Comment by unode** [comment]
My steps:
1. `Manage folders > Show folders > All folders`
2. `Manage folders > Refresh folder list` -> I see (folder names omitted): 

![Screenshot_2020-05-11-20-33-09-657_com fsck k9](https://user-images.githubusercontent.com/122319/81598382-7ccff580-93c7-11ea-9cd1-d452e00b8d18.jpg)

3. `Manage folders > Refresh folder list` clicked a second time and I see:

![Screenshot_2020-05-11-20-33-01-195_com fsck k9](https://user-images.githubusercontent.com/122319/81598398-848f9a00-93c7-11ea-9040-653d44f8defd.jpg)

4. `Manage folders > Refresh folder list` clicked a third time and I see the same as in `2.`

**Comment by cketti** [comment]
Does that match the list of folders retrieved from the server (in the log)? Or is the list in the log identical each time?

**Comment by unode** [comment]
The list of folders in the logs is identical every time.
The only thing that changes is the number of times the messages:
```
K9StorageEditor: Preferences commit took 19 ms
K9StorageEditor: Committing preference changes
```
are shown after the refresh.

**Comment by unode** [comment]
Wondering also if this could be related to #4582 that happened before and the (export/import) workaround I used. Both errors concern the same e-mail account.

**Comment by Virgyxx** [comment]
I have a similar issue

**Comment by cketti** [comment]
I believe I found the bug. Until a new version is released you should be able to use K-9 Mail 5.710 without this issue.

**Comment by unode** [comment]
Thanks!
